app/models/term.rb

```diff
@@ -129,6 +129,9 @@ def retrieve_detection_scores
     # The detections.scores method returns data like [{3=>0.91}, {1=>0.1}] and [{3=>0.95}]
     raw = detections.current.flat_map(&:scores)
     # raw looks like [{3=>0.91}, {1=>0.1}, {3=>0.95}]
+    raw << { @suggested_pattern_category.id => 0.9 } if @suggested_pattern_category.present?
+    raw << { @suggested_resource_category.id => 0.9 } if @suggested_resource_category.present?
```

Comment:
Gotcha... so if I'm following the logic here, for a Term that gets matched by a Suggested* record, even though those do have Detections, the `detections.current.flag_map(&:scores)` line above comes back empty because there's no score - so we use these two lines to make sure that `raw` has what it needs via a different path.

Reply:
Yes, with the minor detail that this _adds_ to raw regardless of what is in there (it's an array).